sources/common/src/TxtParser.cpp

```diff
@@ -60,15 +61,31 @@ void PARSER::TxtParser::Parse()
     std::string input_line;
     while (std::getline(m_LogFile, input_line))
     {
-        const SIGNAL::Signal *signal =
+        const SIGNAL::Signal *pSignal =
             m_rSignalFactory.Create(input_line, m_SourceHandle);
 
-        if (signal)
+        if (pSignal)
         {
-            m_pSignalDb->Add(signal);
+            try
+            {
+                m_pSignalDb->Add(pSignal);
+            }
+            catch (EXCEPTION::VcdException &rException)
+            {
+                if (EXCEPTION::Error::INCONSISTENT_SIGNAL == rException.GetId())
```

Comment:
`delete` should be here. We should consider using `std::unique_ptr` to get rid of such problems.

Reply:
Yes, I thought of it as well. It seems it would be a good opportunity to do it when we will be refactoring the database of signals to save some memory. Well, in the next release.